frontend/src/components/vue_canvas/instances/BoxInstance.ts

```diff
@@ -0,0 +1,342 @@
+import {InstanceBehaviour2D, Instance} from './Instance'
+import {InstanceContext} from './InstanceContext';
+import {v4 as uuidv4} from 'uuid';
+import {getContrastColor} from '../../../utils/colorUtils'
+import {MousePosition, point_is_intersecting_circle} from "../../../types/mouse_position";
+import {get_sequence_color} from '../../regular/regular_annotation'
+import {InstanceColor} from "../../../types/instance_color";
+import {LabelColourMap} from "../../../types/label_colour_map";
+import {ImageCanvasTransform} from "../../../types/CanvasTransform";
+import {InstanceImage2D} from "./InstanceImage2D";
+import {ImageLabelSettings} from "../../../types/image_label_settings";
+
+type BoxHoverPoints =
+  'x_min_y_min'
+  | 'x_max_y_min'
+  | 'x_min_y_max'
+  | 'x_max_y_max'
+  | 'not_intersecting_special_points'
+  | 'blocked'
+
+export class BoxInstance extends InstanceImage2D implements InstanceBehaviour2D {
+  public mouse_position: MousePosition;
+  public ctx: CanvasRenderingContext2D;
+  public canvas_mouse_tools: ImageCanvasTransform;
+  public colour: InstanceColor;
+  public is_dragging_instance: boolean = false;
+  public draw_corners: boolean = false;
+  private is_actively_drawing: boolean = false;
+  public is_moving: boolean = false;
+  public mouse_down_delta_event: any = undefined;
+  public mouse_down_position: any = undefined;
+  public initialized: boolean = false;
+  public box_edit_point_hover: BoxHoverPoints = undefined
+  private nearest_points_dict: any = undefined
+  private zoom_value: number = 1
+  private font_size: number = 10
+
+
+  public get_instance_data(): object {
+    const result = super.get_instance_data();
+    return result;
+  }
+
+  constructor(mouse_position: MousePosition = undefined,
+              ctx: CanvasRenderingContext2D = undefined,
+              on_instance_updated: Function = undefined,
+              on_instance_selected: Function = undefined,
+              on_instance_deselected: Function = undefined,
+              mouse_down_delta_event = undefined,
+              mouse_down_position = undefined,
+              canvas_transform = undefined,
+              image_label_settings: ImageLabelSettings = undefined) {
+
+    super();
+    this.on_instance_updated = on_instance_updated;
+    this.on_instance_selected = on_instance_selected;
+    this.on_instance_deselected = on_instance_deselected;
+    this.mouse_down_delta_event = mouse_down_delta_event;
+    this.mouse_down_position = mouse_down_position;
+    this.canvas_transform = canvas_transform;
+    this.image_label_settings = image_label_settings;
+    this.type = 'box'
+    this.mouse_position = mouse_position;
+    this.initialized = true;
+    this.on_instance_hovered = this.set_default_hover_in_style
+    this.on_instance_unhovered = this.set_default_hover_out_style
+    this.ctx = ctx;
+  }
+
+  public duplicate_for_undo() {
+    let duplicate_instance = new BoxInstance(
+      this.mouse_position,
+      this.ctx,
```

Comment:
I though the idea would be stop using this concept of passing reference to canvas